### ISSUE TITLE
CF_DumpMail Layout

### ISSUE THREAD

**Issue by JamoCA** [kwaschny/cf_dump#10]
This may be outside the scope of this tag, but the current layout (which uses divs) doesn't have a nested appearance like it does when viewed within the browser.

Here are screenshots comparing and contrasting gmail with outlook.

## Gmail
![cf_dump-gmail-20221121140608](https://user-images.githubusercontent.com/1112671/203168446-aa79251b-f9df-4232-9a7d-aedc0a5d2289.png)

## Outlook
![cf_dump-outlook-20221121140750](https://user-images.githubusercontent.com/1112671/203168575-d6130a95-f815-4487-8e7f-2acf484e3571.png)

## Observations
- Gmail's label doesn't have a background color, whereas Outlook is displaying it.
- Gmail has the ability to expand/collapse, Outlook does not.
- Outlook has odd padding/margin. (I don't think "margin" works right w/Outlook.)
- Both don't show nested data as they don't honor the `display: table;` CSS style.

I use [jsoup](https://jsoup.org/) to fix a lot of post-generated HTML.  I tried using a modified version of Ben Nadel's [HTMLEmailUtility](https://github.com/bennadel/Best-Of-ColdFusion-10) project from 11 years ago to "auto-inline" the CSS and the result renders a little closer to the original.  Here's a screenshot of the refactored HTML with the inlined CSS when viewed using gmail. (It didn't affect how it was rendered in Outlook unfortunately.)
![cf_dump-gmail-inline-20221121140631](https://user-images.githubusercontent.com/1112671/203169541-6968e6b3-27d4-46dd-bbdd-cd93d83201cd.png)

Here's how I implemented the CSS inlining using jsoup and Ben's library:
```
<cfsavecontent variable="dumpHtml">
	<cf_dumpmail var="#myVar#" label="cf_dump">
</cfsavecontent>
<cfset inlineCSS = new inlineCSS()>
<cfset dumpHtml = trim(inlineCSS.prepareInline(dumpHtml)>
```

I'm not sure what to recommend, but was caught off guard the first time I tried using it (to replace my CF_Dumplite tag) when a complex struct was outputted as keys on individual lines (instead of nested tables like native CFDump).

**Comment by kwaschny** [comment]
Outlook as in Outlook for Windows? Because `cf_dumpmail` on outlook.com looks fine on our end.

The partial CSS support on web mail clients is really bad and very inconsistent, as you know. I'm not really content with the current markup, but rewriting to tables with HTML4 style attributes? That can't be the solution.

I'll check if I can do something about Gmail.

**Comment by JamoCA** [comment]
We use Office 365, have the Outlook application installed and primarily use it over the web-based version.

I just compared the HTML rendering with outlook.office.com and the web-based version looks outstanding! Much superior to both Gmail and the desktop version of Outlook.

I'm not a fan of using tables in design either, but the HTML support on the Outlook desktop app is rather antiquated.  (My business partner, who does most of the CSS for websites & email messages, has stated that Word is the actual HTML renderer for the desktop app.)  We've use Litmus.com to test and determine what looks best across the most email clients and we often have to add fallback HTML for some email service providers.  (I recently reached out to Disqus as their newsletters weren't even legible as all their content was centered, scrunched & truncated within a 1 inch area with multicolored backgrounds. I didn't want to unsubscribe, but attempting to read it was a futile effort.)

**Comment by kwaschny** [comment]
I moved all styles inline now (12689bf1df0b55a0561ccbf0cf50cf3c0e9be9f5) and that seems to fix almost all issues with GMail. I also took the extra effort of [comparing the computed styles](https://developer.mozilla.org/en-US/docs/Web/API/Window/getComputedStyle) between `<cf_dump>` and `<cf_dumpmail>` to have them look as similar as possible.

I haven't been able to test it against old/native Outlook yet, but it should also improve the situation there.
